File: LeetCode/455. 分发饼干.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort()
        s.sort()
        count = 0
        while g and s:
            if s[0] >= g[0]: 
                count = count + 1
                s.pop(0)
                g.pop(0)
            else:
                s.pop(0)
        return count

File: LeetCode/455. 分发饼干.ts
function findContentChildren(g: number[], s: number[]): number {
    let ans: number = 0;
    g.sort(compare)
    s.sort(compare)
    console.log(g,s);
    for (let i = 0,j = 0;i < g.length && j < s.length;){
        if (g[i] <= s[j]) {
            ans ++;
            i ++;
            j ++;
        }
        else j ++;
    }
    return ans;
};

function compare(a: number, b: number): number{
    // console.log(a,b);
    return a < b?-1:(a==b?0:1);
}